### ISSUE TITLE
Difficultly explaining the expected_value since it '= to mean of predictions on training data

### ISSUE THREAD

**Issue by dwoodlock** [shap/shap#1559]
I'm working on a project to predict length of stay for patients in the hospital so I would like to explain the predictions to the administrators and nurses as part of the system. I feel the need to explain the expected_value properly since shap is defined as impact of features causing the prediction to deviate from the expected_value.  

I am using XGBoost.  I was assuming that the expected_value would be the mean of the predictions of the model against the training data. This is true for regression (hessian = 1 for SSE objective function) but not true for binary classification (my label is 'WillThePatientBeDischargedTodayOrTomorrow') using the negative log likelihood objective function. After walking through the code for a few days and reading xgboost and shap papers, I have figured out shap is using the prediction value at the top of each tree, which essentially is the hessian weighted predictions for the training data for that tree (actually weighted by cover but cover is calculated by the hessian-weighted counts of how many instances went down each part of the tree).

So I could try and explain the expected_value as sort of the mean of the predictions of the training data with some oversampling of the data instances that it was less confident in.  

My questions would be:
- Is this the right definition of the shap methodology or is this an optimization (reading the top of the trees instead of calculating everything)?
- Is there a better explanation of the expected_value that makes sense to domain users?
- Is there a different objective function that I should consider using where the hessian is a constant so it's easier to explain (I don't know if there is one)?

Thanks for your help.
Don
